src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,16 @@
+package duke;
+
+/**
+ * Encapsulates the exception when a parser receives an invalid input.
+ */
+public class DukeException extends Exception {
```

Comment:
If this is meant to only be thrown when an invalid input is given, perhaps a more descriptive name such as `DukeInputException` would help readability?

Reply:
Done. Thanks.